Makefile

```diff
@@ -82,42 +82,49 @@ endif
 
 .PHONY: test
 ifeq ($(TRAVIS),true)
-test: travis
+test: docker-tests
 else
 test: checkstyle
 	OPENQA_CONFIG= prove ${PROVE_ARGS}
 endif
 
-.PHONY: travis
-travis:
-	if test "x$$TRAVIS" != "xtrue"; then \
-	  echo "You shouldn't run this outside of travis-ci env" ;\
-	  exit 1 ;\
-	fi ;\
-	export MOJO_LOG_LEVEL=debug ;\
-	export MOJO_TMPDIR=$$(mktemp -d) ;\
-	export OPENQA_LOGFILE=/tmp/openqa-debug.log ;\
-	export TEST_PG=DBI:Pg:dbname=openqa_test ;\
+.PHONY: docker-tests
+docker-tests:
 	if test "x$$FULLSTACK" = x1 || test "x$$SCHEDULER_FULLSTACK" = x1; then \
 		git clone https://github.com/os-autoinst/os-autoinst.git ../os-autoinst ;\
-		(cd ../os-autoinst && SETUP_FOR_TRAVIS=1 sh autogen.sh && make ) ;\
-					eval $$(dbus-launch --sh-syntax) ;\
-		export PERL5OPT="$$PERL5OPT $$HARNESS_PERL_SWITCHES"; \
+		cd ../os-autoinst ;\
+		cpanm -n --mirror http://no.where/ --installdeps . ;\
+		if [ $$? -eq 0 ]; then\
+			sh autogen.sh && make ;\
+			cd - ;\
+			eval $$(dbus-launch --sh-syntax) ;\
+			export PERL5OPT="$$PERL5OPT $$HARNESS_PERL_SWITCHES" ;\
+		else \
+			echo "OS autoinst dependencies not match. Please check output above" ;\
+			exit 1 ;\
+		fi ;\
 	fi ;\
+	tmp_file=$$(mktemp) ;\
 	if test "x$$FULLSTACK" = x1; then \
-	  perl t/full-stack.t ;\
+	  perl t/full-stack.t | tee $$tmp_file;\
 	elif test "x$$SCHEDULER_FULLSTACK" = x1; then \
-		perl t/05-scheduler-full.t ;\
+		perl t/05-scheduler-full.t | tee $$tmp_file ;\
 	else \
 	  list= ;\
 	  if test "x$$UITESTS" = x1; then \
-	    list=./t/ui ;\
+	    list=$$(find ./t/ui -name *.t | sort ) ;\
 	  else \
 	    $(MAKE) checkstyle ;\
 	    list=$$(find ./t/ -name *.t | grep -v t/ui | sort ) ;\
 	  fi ;\
-          prove ${PROVE_ARGS} --timer -r --jobs 2 $$list ;\
-	fi
+          prove ${PROVE_ARGS} -r $$list | tee $$tmp_file ;\
+	fi ;\
+	tail -n 1 $$tmp_file | grep -o -E "PASS|[[:digit:]]+..[[:digit:]]+" ;\
```

Comment:
Why not better propagate the exit code?

Reply:
Because sometimes it doesn't work.

For example in the test https://travis-ci.org/dasantiago/openQA/jobs/385108697 the test failed, yet returned 0

Reply:
And isn't it a bit scary? :smile:

Reply:
:-D

Reply:
I would say a lot :)

Reply:
@foursixnine 

You know what the output should be in case of success. So evertyhing else is considered failure.